include/veriblock/blockchain/base_block_tree.hpp

```diff
@@ -92,6 +92,13 @@ struct BaseBlockTree {
     }
   }
 
+  void removeTip(index_t& toRemove, bool shouldDetermineBestChain = true) {
+    if (!toRemove.pnext.empty()) {
```

Comment:
```suggestion
    if (!toRemove.isValidTip()) {
```

Block may have next blocks and they may be invalid, so technically current block is tip

Reply:
not from the point of view of rollbacks. those invalid blocks must still be deleted by other UnExecute calls

Reply:
that is, one block per UnExecute